virtcontainers/sandbox.go

```diff
@@ -1364,12 +1363,43 @@ func togglePauseSandbox(sandboxID string, pause bool) (*Sandbox, error) {
 func (s *Sandbox) HotplugAddDevice(device api.Device, devType config.DeviceType) error {
 	switch devType {
 	case config.DeviceVFIO:
-		vfioDevice, ok := device.(*drivers.VFIODevice)
+		vfioDevices, ok := device.GetDeviceDrive().([]*config.VFIODrive)
 		if !ok {
 			return fmt.Errorf("device type mismatch, expect device type to be %s", devType)
 		}
-		_, err := s.hypervisor.hotplugAddDevice(*vfioDevice, vfioDev)
-		return err
+		addedDev := []*config.VFIODrive{}
+		var err error
+		defer func() {
+			// if err happens,roll back and remove added device!
+			if err != nil {
+				for _, dev := range addedDev {
+					if _, rollbackErr := s.hypervisor.hotplugRemoveDevice(dev, vfioDev); rollbackErr != nil {
```

Comment:
Do we care about rolling back such a thing ? I mean the real scenario here is that adding a device will not work for some reasons, and it will return an error. What we expect after this is a `delete` to work, and I don't think trying to remove the device right now or later during the `delete` will help with that.